### ISSUE TITLE
rule proposal: Uses of G_DECLARE_* without a G_DEFINE_*

### ISSUE THREAD

**Issue by A6GibKm** [bilelmoussaoui/gobject-lint#7]
Or the other way around.

E.g. G_DECLARE_FINAL_TYPE was added in 2.44, while G_DEFINE_FINAL_TYPE and other were added in 2.70.

**Comment by bilelmoussaoui** [comment]
Not sure if I understood this correctly but there is now a rule that checks that both macros are consistent between the impl & declaration.

**Comment by A6GibKm** [comment]
I found a case in GTranslator where the G_DECLARE_FINAL_TYPE and G_DEFINE_FINAL_TYPE where both in the .c file. Not sure if that is an issue on the App side tbh.

**Comment by bilelmoussaoui** [comment]
That should be fine, does it trigger a warning here or not?

**Comment by A6GibKm** [comment]
```
gtranslator/src/translation-memory/gtr-translation-memory-ui.c:79:1
  G_DEFINE_FINAL_TYPE (GtrTm, gtr_tm, G_TYPE_OBJECT);
  error: Type 'GtrTm' uses G_DEFINE_FINAL_TYPE but has no corresponding G_DECLARE_FINAL_TYPE in header
  rule: matching_declare_define
```

The code:

```c
#define GTR_TYPE_TM (gtr_tm_get_type ())
G_DECLARE_FINAL_TYPE (GtrTm, gtr_tm, GTR, TM, GObject);

struct _GtrTm {
  GObject parent_instance;
  uint shortcut;
  int level;
  char *match;
};

G_DEFINE_FINAL_TYPE (GtrTm, gtr_tm, G_TYPE_OBJECT);
```

**Comment by bilelmoussaoui** [comment]
I will look into fixing that
